Lib/fontTools/cffLib/CFFToCFF2.py

```diff
@@ -101,9 +102,66 @@ def _convertCFFToCFF2(cff, otFont):
             if hasattr(topDict, key):
                 delattr(topDict, key)
 
+    # Clean up T2CharStrings
+
+    charStrings = topDict.CharStrings
+    globalSubrs = cff.GlobalSubrs
+    localSubrs = [getattr(fd.Private, "Subrs", []) for fd in fdArray]
+
+    for glyphName in charStrings.keys():
+        cs, fdIndex = charStrings.getItemAndSelector(glyphName)
+        cs.decompile()
+
+    # Clean up subroutines first
+    for subrs in [globalSubrs] + localSubrs:
+        for subr in subrs:
+            program = subr.program
+            i = j = len(program)
+            try:
+                i = program.index("return")
+            except ValueError:
+                pass
+            try:
+                j = program.index("endchar")
+            except ValueError:
+                pass
+            program[min(i, j) :] = []
+
+    # Clean up glyph charstrings
+    for glyphName in charStrings.keys():
+        cs, fdIndex = charStrings.getItemAndSelector(glyphName)
+        program = cs.program
+        if fdIndex == None:
+            fdIndex = 0
+
+        # Intentionally use None for nominalWidthX, such that any
+        # CharString that has an explicit width encoded will throw.
+        extractor = T2WidthExtractor(localSubrs[fdIndex], globalSubrs, None, 0)
+        try:
+            extractor.execute(cs)
+        except TypeError:
+            # Program has explicit width. We want to drop it, but can't
+            # just pop the first number since it may be a subroutine call.
+            # Instead, when seeing that, we embed the subroutine and recurse.
+            # This has the problem that some subroutines might become unused.
+            # We don't currently prune those. Subset module has code for this
+            # kind of stuff, possibly plug it in here if pruning becomes needed.
```

Comment:
+1 on moving the subset's subrs pruning to a place where can be reused here

Reply:
https://github.com/fonttools/fonttools/issues/3511